src/collision/CollisionHandler.cpp

```diff
@@ -72,37 +75,102 @@ Entity *CollisionHandler::detectPickUpCollision(std::vector<Entity*> returnObjec
-    int aX, aY, bX, bY;
 
-    std::vector<Entity*> allEntities;
+    auto& zombies = quadtreeZombie.objects;
+    auto& turrets = quadtreeTurret.objects;
+    auto& walls   = quadtreeWall.objects;
+
+    checkTargets(originX, originY, deltaX, deltaY, targetsInSights, zombies, TYPE_ZOMBIE);
+    checkTargets(originX, originY, deltaX, deltaY, targetsInSights, turrets, TYPE_TURRET);
+    checkTargets(originX, originY, deltaX, deltaY, targetsInSights, walls, TYPE_WALL);
+
+    logv("targetsInSights.size(): %d\n", targetsInSights.size());
+}
+
+/**
+    checkTargets
 
-    //split into check for zombies (check for walls)
-    //allEntities = quadtreePro.objects;
+    DISCRIPTION:
+        This checks a single vector of *hitboxes for targets that are in the weapons sights,
+        and adds them to a priority queue that is sorted by distance from the player.
 
-    std::priority_queue<const HitBox*> targetsInSights;
-    for (unsigned int x = 0, len = allEntities.size(); x < len; x++) {
-            aX = playerX;
-            aY = playerY;
-            bX = aX + deltaX;
-            bY = aY + deltaY;
+    AUTHOR: Deric Mccadden 3/16/2017
 
-            if (SDL_IntersectRectAndLine(&(allEntities.at(x)->getProHitBox().getRect()), &aX, &aY , &bX, &bY)) {
-                logv("Shot target at (%d, %d)\n", aX, aY);
-                targetsInSights.push(&(allEntities.at(x)->getProHitBox()));
-            }
+    PARAMS:
+        const int originX,
+            The starting x.
+        const int originY,
+            The starting y.
+        const int deltaX,
+            The x distance to the end point from the origin.
+        const int deltaY,
+            The y distance to the end point from the origin.
+        std::priority_queue<Target>& targetsInSights,
+            This is the queue that will hold the targets that will be determined.
+        std::vector<Entity*> allEntities,
+            The entities that will need to be searched for valid targets.
+        int type
+            the type of entity, see target Target.h for definitions.
+            This is needed for identification purposes in InstantWeapon.fire()
+*/
+void CollisionHandler::checkTargets(const int originX, const int originY, const int deltaX, const int deltaY,
+        std::priority_queue<Target>& targetsInSights, std::vector<Entity*>& allEntities, int type) {
+    int tempOriginX;
+    int tempOriginY;
+    int tempEndX;
+    int tempEndY;
+    int distanceToOrigin;
+
+    for(auto& possibleTarget : allEntities) {
+
+        tempOriginX = originX;
+        tempOriginY = originY;
+        tempEndX = originX + deltaX;
+        tempEndY = originY + deltaY;
+
+        if (SDL_IntersectRectAndLine(&(possibleTarget->getProHitBox().getRect()),
+                &tempOriginX, &tempOriginY , &tempEndX, &tempEndY)) {
+
+            logv("Intersect target at (%d, %d)\n", tempEndX, tempEndY);
+
+            distanceToOrigin = std::hypot(originX - tempEndX, originY - tempEndY);
```

Comment:
Doesn't this hypot caluclation just simplify to deltaX, deltaY ?
originX - originX + deltaX, originY - originY + deltaY
That is redundant